network-config-analyzer/NetworkConfig.py

```diff
@@ -163,33 +152,44 @@ def clone_with_just_one_policy(self, name_of_policy_to_include, policy_type=None
         res.append_policy_to_config(policy)
         return res
 
-    def get_captured_pods(self):
+    def get_captured_pods(self, layer_name=None):
         """
-        :return: All pods captured by any policy
+        :param NetworkLayerName layer_name: The name of a layer to get the pods from
+        :return: All pods captured by any policy, in at least one layer
         :rtype: Peer.PeerSet
         """
+        # TODO: should ignore Ingress layer?
         captured_pods = Peer.PeerSet()
-        for policy in self.sorted_policies:
+        # get all policies list (of all layers) or all policies per input layer
+        if layer_name is None:
+            policies_list = self.policies.values()
+        else:
+            policies_list = self.layers[layer_name].policies_list if layer_name in self.layers else []
+
+        for policy in policies_list:
             captured_pods |= policy.selected_peers
 
-        for profile in self.profiles.values():
-            captured_pods |= profile.selected_peers
+        # TODO:  why was profile.selected_peers considered captured, if profiles may be used for nan captured conns?
+        # for profile in self.profiles.values():
+        #    captured_pods |= profile.selected_peers
```

Comment:
Maybe the name of the function is a bit misleading.
We use this method for optimization: skipping endpoint-pairs for which there cannot be a difference between two network-config because none of them had any connectivity resources that can influence the connections between them. Profiles could have such an effect, and so they were included.

Reply:
ok, updated documentation.